project/Version.scala

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka
+
+import sbt._
+import sbt.Keys._
+
+/**
+ * Generate version.conf and akka/Version.scala files based on the version setting.
+ */
+object Version {
+
+  def versionSettings: Seq[Setting[_]] = inConfig(Compile)(Seq(
+    resourceGenerators <+= generateVersion(resourceManaged, _ / "akka-http-version.conf",
+      """|akka.http.version = "%s"
+         |"""),
+    sourceGenerators <+= generateVersion(sourceManaged, _ / "akka" / "http" / "Version.scala",
+      """|package akka.http
+         |
+         |import com.typesafe.config.Config
+         |
+         |object Version {
+         |  val current: String = "%s"
+         |  def check(config: Config): Unit = {
```

Comment:
Yeah, check is good - imagine if people would force different versions of akka-http and akka-http core - it could break (we are binary compatible externally, but those modules may evolve "together" internally and need exact the same version)